src/ng-pattern-restrict.js

```diff
@@ -217,19 +217,11 @@ angular.module('ngPatternRestrict', [])
             tryParseRegex(entryRegex);
           }
 
-          function notThrows(testFn, shouldReturnTruthy) {
-          	try {
-          		return testFn() || !shouldReturnTruthy;
-          	} catch (e) {
-          		return false;
-          	}
-          }
-
           function detectGetCaretPositionMethods() {
             var input = iElement[0];
-            if (notThrows(function () { return input.selectionStart; })) {
+            if (typeof(input.selectionStart) === 'function') {
               getCaretPosition = getCaretPositionWithInputSelectionStart;
-            } else if (notThrows(function () { return document.selection; }, true)) {
+            } else if (typeof(document.selection) === 'function') {
```

Comment:
document.selection will be an object, not a function. Check how it's used here: https://github.com/udayvarala/ng-pattern-restrict/blob/master/src/ng-pattern-restrict.js#L45

Using it is discouraged and window.getSelection seems to be the option of choice, since this goes for IE9- (http://stackoverflow.com/q/7003784/147507).

Maybe we can check for document.selection being truthy and document.selection.createRange() being a function and moving this all into the "else" of our if.